lib/replicate/checkpointer.js

```diff
@@ -93,8 +161,74 @@ Checkpointer.prototype.getCheckpoint = function () {
     if (err.status !== 404) {
       throw err;
     }
-    return 0;
+    return LOWEST_SEQ;
   });
 };
+// This checkpoint comparison is ported from CouchDBs source
+// they come from here:
+// https://github.com/apache/couchdb-couch-replicator/blob/master/src/couch_replicator.erl#L863-L906
+
+function compareReplicationLogs(srcDoc, tgtDoc) {
+  if(srcDoc.session_id === tgtDoc.session_id) {
+    return {
+      last_seq: srcDoc.last_seq,
+      history: srcDoc.history || []
+    };
+  }
+
+  var sourceHistory = srcDoc.history || [];
+  var targetHistory = tgtDoc.history || [];
+  return compareReplicationHistory(sourceHistory, targetHistory);
+}
+
+function compareReplicationHistory(sourceHistory, targetHistory) {
+  // the erlang loop via function arguments is not so easy to repeat in JS
+  // therefore, doing this as recursion
+  var S = sourceHistory[0];
+  var sourceRest = sourceHistory.slice(1);
+  var T = targetHistory[0];
+  var targetRest = targetHistory.slice(1);
+
+  if(!S || targetHistory.length === 0) {
+    return {
+      last_seq: LOWEST_SEQ,
+      history: []
+    };
+  }
+
+  var sourceId = S.session_id;
+  if(hasSessionId(sourceId, targetHistory)) {
+    return {
+      last_seq: S.last_seq,
```

Comment:
Here and below [`recorded_seq`](https://github.com/apache/couchdb-couch-replicator/blob/master/src/couch_replicator.erl#L891) should be used instead of `last_seq`.